configure.ac

```diff
@@ -54,6 +54,26 @@ AC_TYPE_SIZE_T
 # Checks for library functions.
 AC_CHECK_FUNCS([strpbrk])
 
+STD_CFLAGS="-Wall -Wextra -Wwrite-strings -Wpointer-arith -Wno-missing-field-initializers -Wformat -Wshadow"
+
+# Temporarily disable unused parameter until the implementation is complete
+STD_CFLAGS="$STD_CFLAGS -Wno-unused-parameter"
+
+# These should be always errors
+STD_CFLAGS="$STD_CFLAGS -Werror=implicit-function-declaration -Werror=missing-prototypes -Werror=format-security -Werror=parentheses -Werror=implicit -Werror=strict-prototypes"
+
+# Don't enable warnings on VLA yet, but I would avoid using VLAs
+# STD_CFLAGS"$STD_CFLAGS -Werror=vla"
+
+AX_CHECK_COMPILE_FLAG([-fno-strict-aliasing],
+		      [STD_CFLAGS="$STD_CFLAGS -fno-strict-aliasing"])
+AX_CHECK_COMPILE_FLAG([-Werror -fno-delete-null-pointer-checks],
+		      [STD_CFLAGS="$STD_CFLAGS -fno-delete-null-pointer-checks"])
+AX_CHECK_COMPILE_FLAG([-fdiagnostics-show-option],
+		      [STD_CFLAGS="$STD_CFLAGS -fdiagnostics-show-option"])
+
+AC_SUBST([STD_CFLAGS])
+
```

Comment:
This should stay near the check for c11, either all up to it, or move the c11 cheack down here...

Reply:
Done.